src/exceptions.jl

```diff
@@ -74,9 +74,11 @@ end
 function PQResultError(result::Result; verbose=false)
     msg = error_message(result; verbose=false)
     verbose_msg = verbose ? error_message(result; verbose=true) : nothing
-    code = error_field(result, libpq_c.PG_DIAG_SQLSTATE)
+    code_str = error_field(result, libpq_c.PG_DIAG_SQLSTATE)
+    class = getfield(LibPQ, Symbol("C", code_str[1:2]))
+    code = getfield(LibPQ, Symbol("E", code_str))
```

Comment:
do we want to make these: there own functions?
`parse(ErrorCode, code_str)`
`parse(Class, code_str)`,

related: why not put them in there own submodule?